### PR TITLE
[M21] fix M21 functional bugs

### PR DESCRIPTION
**Pull request by htrajan**
Fix some bugs reported in https://github.com/magefree/mage/issues/6646

### CODE REVIEW

File: Mage.Sets/src/mage/cards/g/GarrukSavageHerald.java [hunk 1]
```diff
@@ -39,15 +45,15 @@ public GarrukSavageHerald(UUID ownerId, CardSetInfo setInfo) {
...
-        DamageWithPowerFromOneToAnotherTargetEffect effect = new DamageWithPowerFromOneToAnotherTargetEffect();
+        Effect effect = new DamageWithPowerFromOneToAnotherTargetEffect();
         effect.setText("Target creature you control deals damage equal to its power to another target creature");
 
         Ability minusAbility = new LoyaltyAbility(effect, -2);
         TargetControlledCreaturePermanent controlledCreature = new TargetControlledCreaturePermanent();
         controlledCreature.setTargetTag(1);
         minusAbility.addTarget(controlledCreature);
 
-        FilterCreaturePermanent filter = new FilterCreaturePermanent("Another creature: damage dealt to");
```
Comment (JayDi85) [inline Mage.Sets/src/mage/cards/g/GarrukSavageHerald.java:50]: As info: there is special method in target to enable extra message in choose dialog for selecting target -- `target.withChooseHint("some hint message")`
shot_200629_072931

Comment (htrajan) [reply Mage.Sets/src/mage/cards/g/GarrukSavageHerald.java:50]: fixed

**Comment by htrajan** [comment]
> 
> 
> The Necromentia fix doesn't seem to have worked, but the other three are solid; marked them off my list.

Is this when testing with Init.txt? It seemed weird when I tried it too (took cards I added to hand and put them in lib when revealing). The code semantically should be correct though.

**Comment by Zerrisx** [comment]
I pre-made decks under test.dck, then necro'd for cards I knew were (or weren't) in the deck.  When I named a card with three copies still in the library, I only got shown the library.  When I named a card with a copy in hand and two in library, I got shown both... and when I named a card not in the deck, I was shown nothing at all; Necromentia just went straight to the yard.

**Comment by htrajan** [comment]
> 
> 
> I pre-made decks under test.dck, then necro'd for cards I knew were (or weren't) in the deck. When I named a card with three copies still in the library, I only got shown the library. When I named a card with a copy in hand and two in library, I got shown both... and when I named a card not in the deck, I was shown nothing at all; Necromentia just went straight to the yard.

Hmm you did pull the latest master branch right? It seems to be working for me. I did the following test with Init.txt:

[init]
battlefield:Human:Swamp:12
hand:Human:Necromentia:3
hand:Computer:Niv-Mizzet Reborn:1
hand:Computer:Savai Triome:4
library:Computer:Savai Triome:3
library:Computer:Counterspell:1
library:Computer:Mountain:6
graveyard:Computer:Savai Triome:2

When I select Savai Triome, I'm prompted to remove any instances of it from their graveyard, hand, and library. For Niv-Mizzet Reborn, I'm asked to choose from hand and it reveals library. For Counterspell, I'm asked to choose from library and it reveals hand.

**Comment by Zerrisx** [comment]
Huh, I recompiled and it's working for me now.  I might just have been sleepy and missed the reveal windows.  Anyway, marking this off as fixed!

**Pull request merged**